packages/camera/camera_avfoundation/example/ios/RunnerTests/CameraSettingsTests.m

```diff
@@ -0,0 +1,244 @@
+}
+
+- (void)testPositiveOrNilShouldAcceptPositiveDecimalNumber {
+  NSError *error;
+  NSNumber *parsed = [CameraPlugin positiveOrNil:@(5) error:&error];
+  XCTAssert(parsed != nil && !error, "should parse positive int number");
+}
+
+@end
+
+@interface CameraSettingsTests : XCTestCase
+@end
+
+@interface TestMediaSettingsProvider : FLTCamMediaSettingsProvider
+@property(nonatomic, readonly) XCTestExpectation *lockExpectation;
+@property(nonatomic, readonly) XCTestExpectation *unlockExpectation;
+@property(nonatomic, readonly) XCTestExpectation *minFrameDurationExpectation;
+@property(nonatomic, readonly) XCTestExpectation *maxFrameDurationExpectation;
+@property(nonatomic, readonly) XCTestExpectation *beginConfigurationExpectation;
+@property(nonatomic, readonly) XCTestExpectation *commitConfigurationExpectation;
+@property(nonatomic, readonly) XCTestExpectation *audioSettingsExpectation;
+@property(nonatomic, readonly) XCTestExpectation *videoSettingsExpectation;
+@end
+
+@implementation TestMediaSettingsProvider
+
+- (instancetype)initWithTestCase:(XCTestCase *)test
+                             fps:(nullable NSNumber *)fps
+                    videoBitrate:(nullable NSNumber *)videoBitrate
+                    audioBitrate:(nullable NSNumber *)audioBitrate
+                     enableAudio:(BOOL)enableAudio {
+  self = [self initWithFps:fps
+              videoBitrate:videoBitrate
+              audioBitrate:audioBitrate
+               enableAudio:enableAudio];
+
+  _lockExpectation = [test expectationWithDescription:@"lockExpectation"];
+  _unlockExpectation = [test expectationWithDescription:@"unlockExpectation"];
+  _minFrameDurationExpectation = [test expectationWithDescription:@"minFrameDurationExpectation"];
+  _maxFrameDurationExpectation = [test expectationWithDescription:@"maxFrameDurationExpectation"];
+  _beginConfigurationExpectation =
+      [test expectationWithDescription:@"beginConfigurationExpectation"];
+  _commitConfigurationExpectation =
+      [test expectationWithDescription:@"commitConfigurationExpectation"];
+  _audioSettingsExpectation = [test expectationWithDescription:@"audioSettingsExpectation"];
+  _videoSettingsExpectation = [test expectationWithDescription:@"videoSettingsExpectation"];
+
+  return self;
+}
+
+- (BOOL)lockDevice:(AVCaptureDevice *)captureDevice error:(NSError **)outError {
+  [_lockExpectation fulfill];
+  return YES;
+}
+
+- (void)unlockDevice:(AVCaptureDevice *)captureDevice {
+  [_unlockExpectation fulfill];
+}
+
+- (void)beginConfigurationForSession:(AVCaptureSession *)videoCaptureSession {
+  [_beginConfigurationExpectation fulfill];
+}
+
+- (void)commitConfigurationForSession:(AVCaptureSession *)videoCaptureSession {
+  [_commitConfigurationExpectation fulfill];
+}
+
+- (void)setMinFrameDuration:(CMTime)duration onDevice:(AVCaptureDevice *)captureDevice {
+  if (duration.value == 10 && duration.timescale == gTestFPS * 10) {
```

Comment:
Where is the magic number 10 coming from here and below?

Reply:
rewritten and commented as:
```objc
  // FLTCam allows to set frame rate with 1/10 precision.
  CMTime expectedDuration = CMTimeMake(10, gTestFPS * 10);
  
  if (duration.value == expectedDuration.value && duration.timescale == expectedDuration.timescale) {
```